ansible_runner/cleanup.py

```diff
@@ -0,0 +1,192 @@
+import glob
+import subprocess
+import os
+import sys
+import signal
+import datetime
+from pathlib import Path
+
+from ansible_runner.defaults import registry_auth_prefix
+from ansible_runner.utils import cleanup_folder
+
+
+__all__ = ['add_cleanup_args', 'run_cleanup']
+
+
+GRACE_PERIOD_DEFAULT = 60  # minutes
+
+
+def add_cleanup_args(command):
+    command.add_argument(
+        "--file-pattern",
+        help="A file glob pattern to find private_data_dir folders to remove. "
+             "Example: --file-pattern=/tmp/.ansible-runner-*"
+    )
+    command.add_argument(
+        "--exclude-strings",
+        nargs='*',
+        help="A comma separated list of keywords in directory name or path to avoid deleting."
+    )
+    command.add_argument(
+        "--remove-images",
+        nargs='*',
+        help="A comma separated list of podman or docker tags to delete. "
+             "This may not remove the corresponding layers, use the image-prune option to assure full deletion. "
+             "Example: --remove-images=quay.io/user/image:devel quay.io/user/builder:latest"
+    )
+    command.add_argument(
+        "--grace-period",
+        default=GRACE_PERIOD_DEFAULT,
+        type=int,
+        help="Time (in minutes) after last modification to exclude a folder from deletion for. "
+             "This is to avoid deleting folders that were recently created, or folders not started via the start command. "
+             "Value of 0 indicates that no folders will be excluded based on modified time."
+    )
+    command.add_argument(
+        "--image-prune",
+        action="store_true",
+        help="If specified, will run docker / podman image prune --force. "
+             "This will only run after untagging."
+    )
+    command.add_argument(
+        "--process-isolation-executable",
+        default="podman",
+        help="The container image to clean up images for (default=podman)"
+    )
+
+
+def run_command(cmd):
```

Comment:
We do in the tests! As for the main code base, I came up a little empty on that search.

Reply:
Ok. I was talking about `run_command()` and `run_command_async()` but those do a bit more than run commands since they also instantiate a `Runner` object.

We should add a generic and simple public function for running commands at a later date.